pkg/istio-agent/sds-agent.go

```diff
@@ -325,14 +329,24 @@ func ingressSdsExists() bool {
 
 // newSecretCache creates the cache for workload secrets and/or gateway secrets.
 func (sa *SDSAgent) newSecretCache(serverOptions sds.Options) (workloadSecretCache *cache.SecretCache, caClient caClientInterface.Client) {
-	ret := &secretfetcher.SecretFetcher{}
+	fetcher := &secretfetcher.SecretFetcher{}
 
 	// TODO: get the MC public keys from pilot.
 	// In node agent, a controller is used getting 'istio-security.istio-system' config map
 	// Single caTLSRootCert inside.
 
 	var err error
 
+	workloadSdsCacheOptions.Plugins = sds.NewPlugins(serverOptions.PluginNames)
+	workloadSecretCache = cache.NewSecretCache(fetcher, sds.NotifyProxy, workloadSdsCacheOptions)
+	sa.WorkloadSecrets = workloadSecretCache
+
+	// If proxy is using file mounted certs, we do not have to connect to CA.
+	if !shouldProvisionCertificates() {
+		log.Info("Workload is using file mounted certificates. Skipping CA provisioning")
```

Comment:
The logging statement says "Skipping CA provisioning" whereas the comment says "we do not have to connect to CA".  It seems that "we do not have to connect to CA" is more accurate in expressing the code change here and thus I suggest revising the logging statement accordingly.